paraatm/bdn/_bdn.py

```diff
@@ -0,0 +1,226 @@
+                       recurrent_regularizer=l2(self.weight_decay),
+                       bias_regularizer=l2(self.weight_decay),dropout=self.idrop,
+                       recurrent_dropout=self.rdrop,input_shape=(None, in_shape),
+                      kernel_initializer='random_uniform',return_sequences=True))
+
+        model.add(Activation('relu'))
+
+        model.add(LSTM(self.num_unit,dropout=self.idrop,
+                       recurrent_dropout=self.rdrop,return_sequences=False,
+                       kernel_regularizer=l2(self.weight_decay),
+                       recurrent_regularizer=l2(self.weight_decay),
+                       bias_regularizer=l2(self.weight_decay)))
+        model.add(Activation('relu'))
+        if self.odrop:
+            model.add(Dropout(self.odrop))
+        model.add(Dense(1,activation='linear',
+                        kernel_regularizer=l2(self.weight_decay),
+                        bias_regularizer=l2(self.weight_decay)))
+        optimizer_=Adam(self.lr)
+        model.compile(loss='mse',metrics=['mse'],optimizer=optimizer_)
+        return model
+    def _get_dnn_model(self):
+        """
+        Construct a deep neural network (dnn) model which is then passed to 'pred' method if model_type =='dnn'.
+        A dnn model is used to make predictions for input data which has linear or non-linear relationships with the output data, or target.
+        
+        
+        For more information about dnn model, please refer to:
+        https://en.wikipedia.org/wiki/Deep_learning#Deep_neural_networks
+        
+        Returns
+        -------
+        A sequential dnn model ready to be fit in 
+        """
+        if len(self.x_train.shape)!=2:
+            raise Exception('expect a 2 dimension array!')
+        in_shape = self.x_train.shape[-1]
+        model=Sequential()
+        model.add(Dense(self.num_unit,kernel_regularizer=l2(self.weight_decay),
+                       bias_regularizer=l2(self.weight_decay),
+                        kernel_initializer='random_uniform',
+                        input_dim=in_shape))
+        
+        model.add(Activation('relu'))
+        model.add(Dropout(self.idrop))
+        model.add(Dense(self.num_unit,kernel_regularizer=l2(self.weight_decay),
+                       bias_regularizer=l2(self.weight_decay),
+                        kernel_initializer='random_uniform',
+                       ))
+        model.add(Dropout(self.idrop))
+        model.add(Activation('relu'))  
+        if self.odrop:
+            model.add(Dropout(self.odrop))
+        model.add(Dense(1,activation='linear',
+                        kernel_regularizer=l2(self.weight_decay),
+                        bias_regularizer=l2(self.weight_decay)))
+        optimizer_=Adam(self.lr)
+        model.compile(loss='mse',metrics=['mse'],optimizer=optimizer_)
+        return model        
+        
+
+    def pred(self, x_test, get_var = True,iter_=5):
```

Comment:
Is `get_var=True` the best default value?  I'm still a little bit unclear on exactly what this does (I think it returns a variance instead of a prediction).  If I understand this correctly, it seems that returning the prediction (`get_var=False`) would be a more logical default.